src/utils/validations/ValidationUtils.ts

```diff
@@ -3,88 +3,74 @@ import { VEHICLE_TYPE, ERRORS } from "../../assets/Enums";
-      return "d";
-    case "other":
-      return "o";
-    case "petrol/oil tanker":
-      return "p";
-    case "skeletal":
-      return "k";
-    case "tipper":
-      return "t";
-    case "box":
-      return "b";
-    case "flat":
-      return "f";
-    case "refuse":
-      return "r";
-    case "skip loader":
-      return "s";
-    case "refrigerated":
-      return "c";
-    default:
-      throw ErrorHandler.Error(400, ERRORS.INVALID_BODY_TYPE);
-  }
+  if (!classCode) throw ErrorHandler.Error(400, ERRORS.INVALID_VEHICLE_CLASS);
+
+  return classCode;
 };
 
-export const populatePartialVin = (vin: string) => (vin.length<6) ? vin : vin.substr(vin.length - 6);
+export const vehicleClassCodeMap = new Map<string, string>([
+  ["3 wheelers",                                     "3"],
+  ["heavy goods vehicle",                            "v"],
+  ["large psv(ie: greater than 23 seats)",           "l"],
+  ["MOT class 4",                                    "4"],
+  ["MOT class 5",                                    "5"],
+  ["MOT class 7",                                    "7"],
+  ["motorbikes over 200cc or with a sidecar",        "2"],
+  ["motorbikes up to 200cc",                         "1"],
+  ["not applicable",                                 "n"],
+  ["small psv (ie: less than or equal to 22 seats)", "s"],
+  ["trailer",                                        "t"]
+]);
+
+export const populateBodyTypeCode = (description: string): string => {
+  const bodyTypeCode = bodyTypeCodeMap.get(description);
+
+  if (!bodyTypeCode) throw ErrorHandler.Error(400, ERRORS.INVALID_BODY_TYPE);
 
-export const populateFields = (techRecord: ITechRecord) => {
+  return bodyTypeCode;
+}
+
+export const bodyTypeCodeMap = new Map<string, string>([
+  ["artic",             "u"],
+  ["articulated",       "a"],
+  ["box",               "b"],
+  ["car transporter",   "y"],
+  ["concrete mixer",    "m"],
+  ["curtainsider",      "e"],
+  ["double decker",     "d"],
+  ["flat",              "f"],
+  ["livestock carrier", "i"],
+  ["low loader",        "l"],
+  ["mini bus",          "m"],
+  ["other",             "o"],
+  ["petrol/oil tanker", "p"],
+  ["refrigerated",      "c"],
+  ["refuse",            "r"],
+  ["single decker",     "s"],
+  ["skeletal",          "k"],
+  ["skip loader",       "s"],
+  ["tipper",            "t"],
+  ["tractor",           "a"]
+]);
+
+export const populatePartialVin = (vin: string): string => vin.length < 6 ? vin : vin.substring(vin.length - 6);
+
+export const populateFields = (techRecord: ITechRecord): void => {
   const { vehicleType } = techRecord;
-  if (
-    vehicleType === VEHICLE_TYPE.PSV ||
-    vehicleType === VEHICLE_TYPE.HGV ||
-    vehicleType === VEHICLE_TYPE.TRL
-  ) {
-    techRecord.bodyType.code = populateBodyTypeCode(
-      techRecord.bodyType.description
-    );
+  
+  if (vehicleType === VEHICLE_TYPE.PSV || vehicleType === VEHICLE_TYPE.HGV || vehicleType === VEHICLE_TYPE.TRL) {
```

Comment:
I feel like this NonLgv check is probably used elsewhere, could we not refactor it into it's own type guard?

Reply:
My code and the scope of the ticket do not alter the functionality of `populateFields()`. I Just removed the indentation here. A quick Shift+Alt+F12 also revealed that this method is not used anywhere in the repo. So, I will leave this as is.